cs_api_cli/cs_api_cli.ml

```diff
@@ -159,17 +222,77 @@ let upload_trace_main
       in
       Cs_api_io.with_client ~config ~f:(fun client ->
           upload_trace ~client ~trace_file ~trace_name ~project_id ~project_name
-            ~api_endpoint ~api_key
+            ~analyze ~api_endpoint ~api_key
           |> Lwt_main.run))
     ~finally:(fun () -> Curl.global_cleanup ())
 
+let analyze_trace_main
+    trace_id
+    profile_id
+    api_endpoint
+    api_key
+    ca_file
+    no_check_certificate =
+  Curl.global_init Curl.CURLINIT_GLOBALALL;
+  Fun.protect
+    (fun () ->
+      let config =
+        {Cs_api_io.Config.verify = not no_check_certificate; ca_file}
+      in
+      let api = Api.make ~api_endpoint ~api_key in
+      Cs_api_io.with_client ~config ~f:(fun client ->
+          analyze_trace ~client ~trace_id ~api ~_rec:false profile_id
+          |> Lwt_main.run))
+    ~finally:(fun () -> Curl.global_cleanup ())
```

Comment:
**suggestion:** Factor out a helper function?

There's a pattern that repeats, here. If we plan to add new commands, maybe it would help if we extracted it as a helper function? Do you think it would be a good idea?

Reply:
Something like this?

Reply:
Looks good to me, indeed. Thanks!